### PR TITLE
feat: activation phase 1

### LINKED ISSUE #24808: [Task]: Feature flag for A/B test
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

Notion: https://www.notion.so/appsmith/Activation-60c64894f42d4cdcb92220c1dbc73802?p=e43ca1f3e1de4fbaa858a51ec2d16571&pm=s

Doc for the A/B test explanation for activation project: https://www.notion.so/appsmith/A-B-testing-in-Activation-Project-e7a3cde48b3d4b5dac7685cd6df5657a?pvs=4

update description after decision: https://theappsmith.slack.com/archives/C056PSE5X0E/p1687524825421209?thread_ts=1687520736.508259&cid=C056PSE5X0E

### LINKED ISSUE #24810: [Feature]: New binding pane
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

Design: https://www.figma.com/file/kbU9xPv44neCfv1FFo9Ndu/User-Activation?type=design&node-id=43-19957&mode=design&t=2qZMFmvPTKakZTEu-0

Notion: https://www.notion.so/appsmith/Activation-60c64894f42d4cdcb92220c1dbc73802?p=7c9a22fa873c4ab9a1a3df7921150398&pm=s

1. The suggested widgets should be shown along with their icons
    1. When I have a successful query, I should see the `Bindings` section as shown in the design (ignore the tooltip/tutorial for this story)
2. Connect to existing widget:
    1. if there are no existing widgets on the canvas, I should see message ‘No existing widgets’ like in the design above
    2. if there are connecting widgets:
        1. we need design for this
        2. [[Open question](https://theappsmith.slack.com/archives/C056PSE5X0E/p1687522136381749?thread_ts=1687520736.508259&cid=C056PSE5X0E)](https://theappsmith.slack.com/archives/C056PSE5X0E/p1687522136381749?thread_ts=1687520736.508259&cid=C056PSE5X0E), please create a new issue if the solution isn’t trivial. For now, leave the existing `Select widget` button as it is. On clicking this, the existing flow should continue, i.e., I should be taken to the canvas and I can snipe to an existing widget.
        
        
        
3. Connect to new widget:
    1. Upon clicking on one of the widgets, the existing snipe flow should kick in, i.e., I should be taken to the canvas and a new widget should be created with the proper data binding from the query response.
4. I should not see the `Relationships` section anymore

### Data

1. Snipe mode used by user
2. Which widget is used in snipe mode
3. A/B test: roughly 50% of users should see this new flow, the remaining should see the old flow
    1. I should be able to see on mixpanel the users who saw the new and old flows

### Why should this be worked on?

for activation project

### LINKED ISSUE #24811: [Feature]: Binding tutorial
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

Design: https://www.figma.com/file/kbU9xPv44neCfv1FFo9Ndu/User-Activation?type=design&node-id=43-19957&mode=design&t=2qZMFmvPTKakZTEu-0

Notion: https://www.notion.so/appsmith/Activation-60c64894f42d4cdcb92220c1dbc73802?p=24f659ce09384beaa1dce433bbd8d02e&pm=s

1. If it is the first time I am successfully executing a query
    1. I should see the tooltip as shown in the design
    2. Along with the gif animation
2. Every subsequent time, I should not see the tutorial if:
    1. I run the query again
    2. I select one of the suggested widgets
    

## Data

1. A/B test: roughly 50% of users should see this new flow, the remaining should see the old flow
    1. I should be able to see on mixpanel the users who saw the new and old flows

### Why should this be worked on?

for activation project

### LINKED ISSUE #24812: [Feature]: Show schema on query page
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

Notion: https://www.notion.so/appsmith/Show-schema-on-query-page-720da69ac71845dab07694c880549a6f

### ### Designs

Loading state:

https://www.figma.com/file/kbU9xPv44neCfv1FFo9Ndu/User-Activation?type=design&node-id=533-56867&mode=design&t=01g0X0pmKoSFDGIL-0

Error state:

https://www.figma.com/file/kbU9xPv44neCfv1FFo9Ndu/User-Activation?type=design&node-id=533-62081&mode=design&t=01g0X0pmKoSFDGIL-0

Working state:

https://www.figma.com/file/kbU9xPv44neCfv1FFo9Ndu/User-Activation?type=design&node-id=533-64974&mode=design&t=01g0X0pmKoSFDGIL-0

Template query:

https://www.figma.com/file/kbU9xPv44neCfv1FFo9Ndu/User-Activation?type=design&node-id=533-62939&mode=design&t=01g0X0pmKoSFDGIL-0

Schema search:

https://www.figma.com/file/kbU9xPv44neCfv1FFo9Ndu/User-Activation?type=design&node-id=533-64439&mode=design&t=01g0X0pmKoSFDGIL-0

### User flow

****************************Loading state([[design](https://www.notion.so/Show-schema-on-query-page-720da69ac71845dab07694c880549a6f?pvs=21)](https://www.notion.so/Show-schema-on-query-page-720da69ac71845dab07694c880549a6f?pvs=21))**

When a user creates a query, the the schema of the underlying datasource by be available(success case) or not(failure case).

In either case, the user should first see a [[loading state](https://www.notion.so/Show-schema-on-query-page-720da69ac71845dab07694c880549a6f?pvs=21)](https://www.notion.so/Show-schema-on-query-page-720da69ac71845dab07694c880549a6f?pvs=21).

************************Failure case([[design](https://www.notion.so/Show-schema-on-query-page-720da69ac71845dab07694c880549a6f?pvs=21)](https://www.notion.so/Show-schema-on-query-page-720da69ac71845dab07694c880549a6f?pvs=21))**

If there is some error in getting the datasource schema, the user should see an error. Also, considering that in most error cases the datasource configuration is invalid, the user should also see a cta to edit the datasource, upon clicking this, they should be taken to the datasource edit form.

****************Success case([[design](https://www.notion.so/Show-schema-on-query-page-720da69ac71845dab07694c880549a6f?pvs=21)](https://www.notion.so/Show-schema-on-query-page-720da69ac71845dab07694c880549a6f?pvs=21))**

When the schema is available, the user should see it as shown in the design.

**Template query([[design](https://www.notion.so/Show-schema-on-query-page-720da69ac71845dab07694c880549a6f?pvs=21)](https://www.notion.so/Show-schema-on-query-page-720da69ac71845dab07694c880549a6f?pvs=21))**

The user can also make use of a template query as shown in the design, doing this should work exactly as `smart templates` work today. i.e., a new query should be created with the generated template query.

****************************Search([[design](https://www.notion.so/Show-schema-on-query-page-720da69ac71845dab07694c880549a6f?pvs=21)](https://www.notion.so/Show-schema-on-query-page-720da69ac71845dab07694c880549a6f?pvs=21))**

The user should also be able to search in the schema as shown in the design.

### Data

1. number of times schema fetch failed/passed
2. number of times the user chose a template query
3. A/B test: roughly 50% of users should see this new flow, the remaining should see the old flow
    1. I should be able to see on mixpanel the users who saw the new and old flows

### Why should this be worked on?

for activation project

### LINKED ISSUE #25242: [Feature]: Automatic Query generation with Undo Functionality
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

This feature allows users to automatically generate a query from the query editor page and have the query be populated in the editor fields without creating a new query (like what happens in the entity explorer). Users should also be able to undo whatever changes they have made.

### Why should this be worked on?

This should be worked on as part of the activation project.

### LINKED ISSUE #25300: [Task]: Analytics Event for Automatic Query Generation
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Summary

The event should be called AUTOMATIC_QUERY_GENERATION,
And the information being tracked should be

1. the datasourceId
2. the plugin name for that query (MongoDB, MySQL etc.)
3. the template title for the generated template (SELECT, INSERT - for SQL queries or FIND, FIND BY ID -  for Mongo etc.)
4. whether the walkthrough is opened or not.

### Why should this be worked on?

As part of the Activation project.

### PR DESCRIPTION
**Pull request by Irongade**
Problem Statement:

https://www.notion.so/appsmith/Activation-60c64894f42d4cdcb92220c1dbc73802

- New feature (non-breaking change which adds functionality)

#### PR fixes following issue(s)
Fixes #24810 
Fixes #24812 
Fixes #24811  
Fixes #24808 
Fixes #25199 
Fixes #25242 
Fixes #25300 

- [ ] Manual
- [ ] Jest
- [x] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/cypress/e2e/Regression/ClientSide/BugTests/ApiBugs_Spec.ts [hunk 1]
```diff
@@ -15,6 +15,12 @@ import {
 } from "../../../../support/Objects/CommonErrorMessages";
 
 describe("API Bugs", function () {
+  before(() => {
+    cy.intercept("GET", "/api/v1/users/features", {
```
Comment (hetunandu) [inline app/client/cypress/e2e/Regression/ClientSide/BugTests/ApiBugs_Spec.ts:19]: We are changing this api. Please look at https://github.com/appsmithorg/appsmith/pull/25104 for more info.

File: app/client/src/components/editorComponents/ActionRightPane/SuggestedWidgets.tsx [hunk 2]
```diff
@@ -20,8 +27,41 @@ import { getNextWidgetName } from "sagas/WidgetOperationUtils";
...
+import WalkthroughContext from "components/featureWalkthrough/walkthroughContext";
+import {
+  getFeatureFlagShownStatus,
+  isUserSignedUpFlagSet,
+  setFeatureFlagShownStatus,
+} from "utils/storage";
+import { getCurrentUser } from "selectors/usersSelectors";
+
+const BINDING_GUIDE_GIF =
+  "https://s3.us-east-2.amazonaws.com/assets.appsmith.com/binding.gif";
```
Comment (sneha122) [inline app/client/src/components/editorComponents/ActionRightPane/SuggestedWidgets.tsx:58]: @AmanAgarwal041 Can we please replace this path with `${ASSETS_CDN_URL}/binding.gif`? Please check line 174 in same file for reference.

File: app/client/src/components/editorComponents/ActionRightPane/index.tsx [hunk 3]
```diff
@@ -31,16 +35,39 @@ import {
...
+  getFeatureFlagShownStatus,
+  isUserSignedUpFlagSet,
+  setFeatureFlagShownStatus,
+} from "utils/storage";
+import { PluginName } from "entities/Action";
+import { getCurrentUser } from "selectors/usersSelectors";
+import { Tooltip } from "design-system";
+
+const SCHEMA_GUIDE_GIF =
+  "https://s3.us-east-2.amazonaws.com/assets.appsmith.com/schema.gif";
```
Comment (sneha122) [inline app/client/src/components/editorComponents/ActionRightPane/index.tsx:64]: @AmanAgarwal041 Same here, can we please use `${ASSETS_CDN_URL}/schema.gif`

**Review by AmanAgarwal041 — APPROVED** [review]

**Review by sneha122 — APPROVED** [review]

**Pull request merged**